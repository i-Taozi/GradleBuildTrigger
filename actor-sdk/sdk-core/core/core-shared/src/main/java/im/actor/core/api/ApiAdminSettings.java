package im.actor.core.api;
/*
 *  Generated by the Actor API Scheme generator.  DO NOT EDIT!
 */

import im.actor.runtime.bser.*;
import im.actor.runtime.collections.*;
import static im.actor.runtime.bser.Utils.*;
import im.actor.core.network.parser.*;
import org.jetbrains.annotations.Nullable;
import org.jetbrains.annotations.NotNull;
import com.google.j2objc.annotations.ObjectiveCName;
import java.io.IOException;
import java.util.List;
import java.util.ArrayList;

public class ApiAdminSettings extends BserObject {

    private boolean showAdminsToMembers;
    private boolean canMembersInvite;
    private boolean canMembersEditGroupInfo;
    private boolean canAdminsEditGroupInfo;
    private boolean showJoinLeaveMessages;

    public ApiAdminSettings(boolean showAdminsToMembers, boolean canMembersInvite, boolean canMembersEditGroupInfo, boolean canAdminsEditGroupInfo, boolean showJoinLeaveMessages) {
        this.showAdminsToMembers = showAdminsToMembers;
        this.canMembersInvite = canMembersInvite;
        this.canMembersEditGroupInfo = canMembersEditGroupInfo;
        this.canAdminsEditGroupInfo = canAdminsEditGroupInfo;
        this.showJoinLeaveMessages = showJoinLeaveMessages;
    }

    public ApiAdminSettings() {

    }

    public boolean showAdminsToMembers() {
        return this.showAdminsToMembers;
    }

    public boolean canMembersInvite() {
        return this.canMembersInvite;
    }

    public boolean canMembersEditGroupInfo() {
        return this.canMembersEditGroupInfo;
    }

    public boolean canAdminsEditGroupInfo() {
        return this.canAdminsEditGroupInfo;
    }

    public boolean showJoinLeaveMessages() {
        return this.showJoinLeaveMessages;
    }

    @Override
    public void parse(BserValues values) throws IOException {
        this.showAdminsToMembers = values.getBool(1);
        this.canMembersInvite = values.getBool(2);
        this.canMembersEditGroupInfo = values.getBool(3);
        this.canAdminsEditGroupInfo = values.getBool(4);
        this.showJoinLeaveMessages = values.getBool(5);
        if (values.hasRemaining()) {
            setUnmappedObjects(values.buildRemaining());
        }
    }

    @Override
    public void serialize(BserWriter writer) throws IOException {
        writer.writeBool(1, this.showAdminsToMembers);
        writer.writeBool(2, this.canMembersInvite);
        writer.writeBool(3, this.canMembersEditGroupInfo);
        writer.writeBool(4, this.canAdminsEditGroupInfo);
        writer.writeBool(5, this.showJoinLeaveMessages);
        if (this.getUnmappedObjects() != null) {
            SparseArray<Object> unmapped = this.getUnmappedObjects();
            for (int i = 0; i < unmapped.size(); i++) {
                int key = unmapped.keyAt(i);
                writer.writeUnmapped(key, unmapped.get(key));
            }
        }
    }

    @Override
    public String toString() {
        String res = "struct AdminSettings{";
        res += "showAdminsToMembers=" + this.showAdminsToMembers;
        res += ", canMembersInvite=" + this.canMembersInvite;
        res += ", canMembersEditGroupInfo=" + this.canMembersEditGroupInfo;
        res += ", canAdminsEditGroupInfo=" + this.canAdminsEditGroupInfo;
        res += ", showJoinLeaveMessages=" + this.showJoinLeaveMessages;
        res += "}";
        return res;
    }

}
