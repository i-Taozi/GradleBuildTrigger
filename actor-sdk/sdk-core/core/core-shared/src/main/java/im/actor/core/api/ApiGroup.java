package im.actor.core.api;
/*
 *  Generated by the Actor API Scheme generator.  DO NOT EDIT!
 */

import im.actor.runtime.bser.*;
import im.actor.runtime.collections.*;
import static im.actor.runtime.bser.Utils.*;
import im.actor.core.network.parser.*;
import org.jetbrains.annotations.Nullable;
import org.jetbrains.annotations.NotNull;
import com.google.j2objc.annotations.ObjectiveCName;
import java.io.IOException;
import java.util.List;
import java.util.ArrayList;

public class ApiGroup extends BserObject {

    private int id;
    private long accessHash;
    private String title;
    private ApiAvatar avatar;
    private Integer membersCount;
    private Boolean isMember;
    private Boolean isHidden;
    private ApiGroupType groupType;
    private Long permissions;
    private Boolean isDeleted;
    private ApiMapValue ext;

    public ApiGroup(int id, long accessHash, @NotNull String title, @Nullable ApiAvatar avatar, @Nullable Integer membersCount, @Nullable Boolean isMember, @Nullable Boolean isHidden, @Nullable ApiGroupType groupType, @Nullable Long permissions, @Nullable Boolean isDeleted, @Nullable ApiMapValue ext) {
        this.id = id;
        this.accessHash = accessHash;
        this.title = title;
        this.avatar = avatar;
        this.membersCount = membersCount;
        this.isMember = isMember;
        this.isHidden = isHidden;
        this.groupType = groupType;
        this.permissions = permissions;
        this.isDeleted = isDeleted;
        this.ext = ext;
    }

    public ApiGroup() {

    }

    public int getId() {
        return this.id;
    }

    public long getAccessHash() {
        return this.accessHash;
    }

    @NotNull
    public String getTitle() {
        return this.title;
    }

    @Nullable
    public ApiAvatar getAvatar() {
        return this.avatar;
    }

    @Nullable
    public Integer getMembersCount() {
        return this.membersCount;
    }

    @Nullable
    public Boolean isMember() {
        return this.isMember;
    }

    @Nullable
    public Boolean isHidden() {
        return this.isHidden;
    }

    @Nullable
    public ApiGroupType getGroupType() {
        return this.groupType;
    }

    @Nullable
    public Long getPermissions() {
        return this.permissions;
    }

    @Nullable
    public Boolean isDeleted() {
        return this.isDeleted;
    }

    @Nullable
    public ApiMapValue getExt() {
        return this.ext;
    }

    @Override
    public void parse(BserValues values) throws IOException {
        this.id = values.getInt(1);
        this.accessHash = values.getLong(2);
        this.title = values.getString(3);
        this.avatar = values.optObj(4, new ApiAvatar());
        this.membersCount = values.optInt(24);
        this.isMember = values.optBool(6);
        this.isHidden = values.optBool(20);
        int val_groupType = values.getInt(25, 0);
        if (val_groupType != 0) {
            this.groupType = ApiGroupType.parse(val_groupType);
        }
        this.permissions = values.optLong(26);
        this.isDeleted = values.optBool(27);
        this.ext = values.optObj(22, new ApiMapValue());
        if (values.hasRemaining()) {
            setUnmappedObjects(values.buildRemaining());
        }
    }

    @Override
    public void serialize(BserWriter writer) throws IOException {
        writer.writeInt(1, this.id);
        writer.writeLong(2, this.accessHash);
        if (this.title == null) {
            throw new IOException();
        }
        writer.writeString(3, this.title);
        if (this.avatar != null) {
            writer.writeObject(4, this.avatar);
        }
        if (this.membersCount != null) {
            writer.writeInt(24, this.membersCount);
        }
        if (this.isMember != null) {
            writer.writeBool(6, this.isMember);
        }
        if (this.isHidden != null) {
            writer.writeBool(20, this.isHidden);
        }
        if (this.groupType != null) {
            writer.writeInt(25, this.groupType.getValue());
        }
        if (this.permissions != null) {
            writer.writeLong(26, this.permissions);
        }
        if (this.isDeleted != null) {
            writer.writeBool(27, this.isDeleted);
        }
        if (this.ext != null) {
            writer.writeObject(22, this.ext);
        }
        if (this.getUnmappedObjects() != null) {
            SparseArray<Object> unmapped = this.getUnmappedObjects();
            for (int i = 0; i < unmapped.size(); i++) {
                int key = unmapped.keyAt(i);
                writer.writeUnmapped(key, unmapped.get(key));
            }
        }
    }

    @Override
    public String toString() {
        String res = "struct Group{";
        res += "id=" + this.id;
        res += ", title=" + this.title;
        res += ", avatar=" + (this.avatar != null ? "set":"empty");
        res += ", membersCount=" + this.membersCount;
        res += ", isMember=" + this.isMember;
        res += ", isHidden=" + this.isHidden;
        res += ", groupType=" + this.groupType;
        res += ", permissions=" + this.permissions;
        res += ", isDeleted=" + this.isDeleted;
        res += ", ext=" + this.ext;
        res += "}";
        return res;
    }

}
