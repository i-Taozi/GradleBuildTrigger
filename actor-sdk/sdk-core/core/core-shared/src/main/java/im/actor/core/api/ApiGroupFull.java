package im.actor.core.api;
/*
 *  Generated by the Actor API Scheme generator.  DO NOT EDIT!
 */

import im.actor.runtime.bser.*;
import im.actor.runtime.collections.*;
import static im.actor.runtime.bser.Utils.*;
import im.actor.core.network.parser.*;
import org.jetbrains.annotations.Nullable;
import org.jetbrains.annotations.NotNull;
import com.google.j2objc.annotations.ObjectiveCName;
import java.io.IOException;
import java.util.List;
import java.util.ArrayList;

public class ApiGroupFull extends BserObject {

    private int id;
    private long createDate;
    private Integer ownerUid;
    private List<ApiMember> members;
    private String theme;
    private String about;
    private ApiMapValue ext;
    private Boolean isAsyncMembers;
    private Boolean isSharedHistory;
    private String shortName;
    private Long permissions;

    public ApiGroupFull(int id, long createDate, @Nullable Integer ownerUid, @NotNull List<ApiMember> members, @Nullable String theme, @Nullable String about, @Nullable ApiMapValue ext, @Nullable Boolean isAsyncMembers, @Nullable Boolean isSharedHistory, @Nullable String shortName, @Nullable Long permissions) {
        this.id = id;
        this.createDate = createDate;
        this.ownerUid = ownerUid;
        this.members = members;
        this.theme = theme;
        this.about = about;
        this.ext = ext;
        this.isAsyncMembers = isAsyncMembers;
        this.isSharedHistory = isSharedHistory;
        this.shortName = shortName;
        this.permissions = permissions;
    }

    public ApiGroupFull() {

    }

    public int getId() {
        return this.id;
    }

    public long getCreateDate() {
        return this.createDate;
    }

    @Nullable
    public Integer getOwnerUid() {
        return this.ownerUid;
    }

    @NotNull
    public List<ApiMember> getMembers() {
        return this.members;
    }

    @Nullable
    public String getTheme() {
        return this.theme;
    }

    @Nullable
    public String getAbout() {
        return this.about;
    }

    @Nullable
    public ApiMapValue getExt() {
        return this.ext;
    }

    @Nullable
    public Boolean isAsyncMembers() {
        return this.isAsyncMembers;
    }

    @Nullable
    public Boolean isSharedHistory() {
        return this.isSharedHistory;
    }

    @Nullable
    public String getShortName() {
        return this.shortName;
    }

    @Nullable
    public Long getPermissions() {
        return this.permissions;
    }

    @Override
    public void parse(BserValues values) throws IOException {
        this.id = values.getInt(1);
        this.createDate = values.getLong(6);
        this.ownerUid = values.optInt(5);
        List<ApiMember> _members = new ArrayList<ApiMember>();
        for (int i = 0; i < values.getRepeatedCount(12); i ++) {
            _members.add(new ApiMember());
        }
        this.members = values.getRepeatedObj(12, _members);
        this.theme = values.optString(2);
        this.about = values.optString(3);
        this.ext = values.optObj(7, new ApiMapValue());
        this.isAsyncMembers = values.optBool(11);
        this.isSharedHistory = values.optBool(10);
        this.shortName = values.optString(14);
        this.permissions = values.optLong(27);
        if (values.hasRemaining()) {
            setUnmappedObjects(values.buildRemaining());
        }
    }

    @Override
    public void serialize(BserWriter writer) throws IOException {
        writer.writeInt(1, this.id);
        writer.writeLong(6, this.createDate);
        if (this.ownerUid != null) {
            writer.writeInt(5, this.ownerUid);
        }
        writer.writeRepeatedObj(12, this.members);
        if (this.theme != null) {
            writer.writeString(2, this.theme);
        }
        if (this.about != null) {
            writer.writeString(3, this.about);
        }
        if (this.ext != null) {
            writer.writeObject(7, this.ext);
        }
        if (this.isAsyncMembers != null) {
            writer.writeBool(11, this.isAsyncMembers);
        }
        if (this.isSharedHistory != null) {
            writer.writeBool(10, this.isSharedHistory);
        }
        if (this.shortName != null) {
            writer.writeString(14, this.shortName);
        }
        if (this.permissions != null) {
            writer.writeLong(27, this.permissions);
        }
        if (this.getUnmappedObjects() != null) {
            SparseArray<Object> unmapped = this.getUnmappedObjects();
            for (int i = 0; i < unmapped.size(); i++) {
                int key = unmapped.keyAt(i);
                writer.writeUnmapped(key, unmapped.get(key));
            }
        }
    }

    @Override
    public String toString() {
        String res = "struct GroupFull{";
        res += "id=" + this.id;
        res += ", createDate=" + this.createDate;
        res += ", ownerUid=" + this.ownerUid;
        res += ", members=" + this.members;
        res += ", theme=" + this.theme;
        res += ", about=" + this.about;
        res += ", isAsyncMembers=" + this.isAsyncMembers;
        res += ", isSharedHistory=" + this.isSharedHistory;
        res += ", shortName=" + this.shortName;
        res += ", permissions=" + this.permissions;
        res += "}";
        return res;
    }

}
