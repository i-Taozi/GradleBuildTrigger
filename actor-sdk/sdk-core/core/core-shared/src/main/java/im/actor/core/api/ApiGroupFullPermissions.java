package im.actor.core.api;
/*
 *  Generated by the Actor API Scheme generator.  DO NOT EDIT!
 */

import java.io.IOException;

public enum ApiGroupFullPermissions {

    EDIT_INFO(1),
    VIEW_MEMBERS(2),
    INVITE_MEMBERS(3),
    INVITE_VIA_LINK(4),
    CALL(5),
    EDIT_ADMIN_SETTINGS(6),
    VIEW_ADMINS(7),
    EDIT_ADMINS(8),
    KICK_INVITED(9),
    KICK_ANYONE(10),
    EDIT_FOREIGN(11),
    DELETE_FOREIGN(12),
    UNSUPPORTED_VALUE(-1);

    private int value;

    ApiGroupFullPermissions(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public static ApiGroupFullPermissions parse(int value) throws IOException {
        switch(value) {
            case 1: return ApiGroupFullPermissions.EDIT_INFO;
            case 2: return ApiGroupFullPermissions.VIEW_MEMBERS;
            case 3: return ApiGroupFullPermissions.INVITE_MEMBERS;
            case 4: return ApiGroupFullPermissions.INVITE_VIA_LINK;
            case 5: return ApiGroupFullPermissions.CALL;
            case 6: return ApiGroupFullPermissions.EDIT_ADMIN_SETTINGS;
            case 7: return ApiGroupFullPermissions.VIEW_ADMINS;
            case 8: return ApiGroupFullPermissions.EDIT_ADMINS;
            case 9: return ApiGroupFullPermissions.KICK_INVITED;
            case 10: return ApiGroupFullPermissions.KICK_ANYONE;
            case 11: return ApiGroupFullPermissions.EDIT_FOREIGN;
            case 12: return ApiGroupFullPermissions.DELETE_FOREIGN;
            default: return ApiGroupFullPermissions.UNSUPPORTED_VALUE;
        }
    }
}
