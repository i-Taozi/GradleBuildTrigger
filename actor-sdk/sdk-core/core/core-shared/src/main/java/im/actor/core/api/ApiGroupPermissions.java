package im.actor.core.api;
/*
 *  Generated by the Actor API Scheme generator.  DO NOT EDIT!
 */

import java.io.IOException;

public enum ApiGroupPermissions {

    SEND_MESSAGE(1),
    CLEAR(2),
    LEAVE(3),
    DELETE(4),
    JOIN(5),
    VIEW_INFO(6),
    UNSUPPORTED_VALUE(-1);

    private int value;

    ApiGroupPermissions(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public static ApiGroupPermissions parse(int value) throws IOException {
        switch(value) {
            case 1: return ApiGroupPermissions.SEND_MESSAGE;
            case 2: return ApiGroupPermissions.CLEAR;
            case 3: return ApiGroupPermissions.LEAVE;
            case 4: return ApiGroupPermissions.DELETE;
            case 5: return ApiGroupPermissions.JOIN;
            case 6: return ApiGroupPermissions.VIEW_INFO;
            default: return ApiGroupPermissions.UNSUPPORTED_VALUE;
        }
    }
}
