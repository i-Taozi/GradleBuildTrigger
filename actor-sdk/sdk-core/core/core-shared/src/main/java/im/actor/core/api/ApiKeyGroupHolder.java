package im.actor.core.api;
/*
 *  Generated by the Actor API Scheme generator.  DO NOT EDIT!
 */

import im.actor.runtime.bser.*;
import im.actor.runtime.collections.*;
import static im.actor.runtime.bser.Utils.*;
import im.actor.core.network.parser.*;
import org.jetbrains.annotations.Nullable;
import org.jetbrains.annotations.NotNull;
import com.google.j2objc.annotations.ObjectiveCName;
import java.io.IOException;
import java.util.List;
import java.util.ArrayList;

public class ApiKeyGroupHolder extends BserObject {

    private int uid;
    private ApiEncryptionKeyGroup keyGroup;

    public ApiKeyGroupHolder(int uid, @NotNull ApiEncryptionKeyGroup keyGroup) {
        this.uid = uid;
        this.keyGroup = keyGroup;
    }

    public ApiKeyGroupHolder() {

    }

    public int getUid() {
        return this.uid;
    }

    @NotNull
    public ApiEncryptionKeyGroup getKeyGroup() {
        return this.keyGroup;
    }

    @Override
    public void parse(BserValues values) throws IOException {
        this.uid = values.getInt(1);
        this.keyGroup = values.getObj(2, new ApiEncryptionKeyGroup());
    }

    @Override
    public void serialize(BserWriter writer) throws IOException {
        writer.writeInt(1, this.uid);
        if (this.keyGroup == null) {
            throw new IOException();
        }
        writer.writeObject(2, this.keyGroup);
    }

    @Override
    public String toString() {
        String res = "struct KeyGroupHolder{";
        res += "uid=" + this.uid;
        res += ", keyGroup=" + this.keyGroup;
        res += "}";
        return res;
    }

}
