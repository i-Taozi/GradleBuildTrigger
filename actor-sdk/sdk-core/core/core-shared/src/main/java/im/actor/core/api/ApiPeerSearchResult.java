package im.actor.core.api;
/*
 *  Generated by the Actor API Scheme generator.  DO NOT EDIT!
 */

import im.actor.runtime.bser.*;
import im.actor.runtime.collections.*;
import static im.actor.runtime.bser.Utils.*;
import im.actor.core.network.parser.*;
import org.jetbrains.annotations.Nullable;
import org.jetbrains.annotations.NotNull;
import com.google.j2objc.annotations.ObjectiveCName;
import java.io.IOException;
import java.util.List;
import java.util.ArrayList;

public class ApiPeerSearchResult extends BserObject {

    private ApiPeer peer;
    private String optMatchString;

    public ApiPeerSearchResult(@NotNull ApiPeer peer, @Nullable String optMatchString) {
        this.peer = peer;
        this.optMatchString = optMatchString;
    }

    public ApiPeerSearchResult() {

    }

    @NotNull
    public ApiPeer getPeer() {
        return this.peer;
    }

    @Nullable
    public String getOptMatchString() {
        return this.optMatchString;
    }

    @Override
    public void parse(BserValues values) throws IOException {
        this.peer = values.getObj(1, new ApiPeer());
        this.optMatchString = values.optString(3);
    }

    @Override
    public void serialize(BserWriter writer) throws IOException {
        if (this.peer == null) {
            throw new IOException();
        }
        writer.writeObject(1, this.peer);
        if (this.optMatchString != null) {
            writer.writeString(3, this.optMatchString);
        }
    }

    @Override
    public String toString() {
        String res = "struct PeerSearchResult{";
        res += "peer=" + this.peer;
        res += ", optMatchString=" + this.optMatchString;
        res += "}";
        return res;
    }

}
