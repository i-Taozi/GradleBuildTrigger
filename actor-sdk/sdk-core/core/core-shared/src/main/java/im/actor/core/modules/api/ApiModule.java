package im.actor.core.modules.api;

import java.io.IOException;

import im.actor.core.modules.AbsModule;
import im.actor.core.modules.Modules;
import im.actor.core.events.AppVisibleChanged;
import im.actor.core.events.ConnectingStateChanged;
import im.actor.core.events.NewSessionCreated;
import im.actor.core.network.ActorApi;
import im.actor.core.network.ActorApiCallback;
import im.actor.core.network.AuthKeyStorage;
import im.actor.core.network.Endpoints;
import im.actor.core.network.TrustedKey;
import im.actor.core.network.parser.Request;
import im.actor.runtime.actors.ActorRef;
import im.actor.runtime.eventbus.BusSubscriber;
import im.actor.runtime.eventbus.Event;
import im.actor.runtime.mtproto.ConnectionEndpoint;
import im.actor.runtime.mtproto.ConnectionEndpointArray;
import im.actor.runtime.promise.Promise;

import static im.actor.runtime.actors.ActorSystem.system;

public class ApiModule extends AbsModule implements BusSubscriber {

    private final ActorApi actorApi;
    private final ActorRef persistentRequests;
    private final AuthKeyStorage authKeyStorage;

    public ApiModule(Modules context) {
        super(context);

        this.authKeyStorage = new PreferenceApiStorage(context().getPreferences());

        Endpoints endpoints = null;
        byte[] customEndpointsBytes = context().getPreferences().getBytes("custom_endpoints");
        if (customEndpointsBytes != null) {
            try {
                endpoints = Endpoints.fromBytes(customEndpointsBytes);
            } catch (IOException e) {
                e.printStackTrace();
            }
        }

        if (endpoints == null) {
            endpoints = new Endpoints(context().getConfiguration().getEndpoints(), context().getConfiguration().getTrustedKeys());
        }

        this.actorApi = new ActorApi(endpoints,
                authKeyStorage,
                new ActorApiCallbackImpl(),
                context().getConfiguration().isEnableNetworkLogging(),
                context().getConfiguration().getMinDelay(),
                context().getConfiguration().getMaxDelay(),
                context().getConfiguration().getMaxFailureCount());

        context.getEvents().subscribe(this, AppVisibleChanged.EVENT);

        persistentRequests = system().actorOf("api/persistence", () -> {
            return new PersistentRequestsActor(context());
        });
    }

    /**
     * Get Actor API instance
     *
     * @return Actor API instance
     */
    public ActorApi getActorApi() {
        return actorApi;
    }

    /**
     * DANGER. Get Auth Keys storage. Do not use it if you don't understand what it is.
     *
     * @return Auth Key Storage
     */
    public AuthKeyStorage getAuthKeyStorage() {
        return authKeyStorage;
    }

    /**
     * Performing persist request. Keep repeating request even after application restart
     *
     * @param request request
     */
    public void performPersistRequest(Request request) {
        persistentRequests.send(new PersistentRequestsActor.PerformRequest(request));
    }

    /**
     * Perform cursor persist request. Request is performed only if key is bigger than previous
     * request with same name
     *
     * @param name    name of cursor
     * @param key     sorting key
     * @param request request for performing
     */
    public void performPersistCursorRequest(String name, long key, Request request) {
        persistentRequests.send(new PersistentRequestsActor.PerformCursorRequest(name, key, request));
    }

    /**
     * Changing endpoint
     */
    public void changeEndpoint(String endpoint) throws ConnectionEndpointArray.UnknownSchemeException {
        changeEndpoints(new Endpoints(new ConnectionEndpointArray().addEndpoint(endpoint).toArray(new ConnectionEndpoint[1]), new TrustedKey[0]));
    }

    /**
     * Changing endpoints
     */
    public synchronized void changeEndpoints(Endpoints endpoints) {
        context().getPreferences().putBytes("custom_endpoints", endpoints.toByteArray());
        actorApi.changeEndpoints(endpoints);
    }

    /**
     * Reset default endpoints
     */
    public synchronized void resetToDefaultEndpoints() {
        context().getPreferences().putBytes("custom_endpoints", null);
        actorApi.resetToDefaultEndpoints();
    }

    public Promise<Boolean> checkIsCurrentAuthId(long authId) {
        return actorApi.checkIsCurrentAuthId(authId);
    }

    @Override
    public void onBusEvent(Event event) {
        if (event instanceof AppVisibleChanged) {
            if (((AppVisibleChanged) event).isVisible()) {
                actorApi.forceNetworkCheck();
            }
        }
    }

    private class ActorApiCallbackImpl implements ActorApiCallback {

        @Override
        public void onAuthIdInvalidated() {
            ((Modules) context()).onLoggedOut();
        }

        @Override
        public void onNewSessionCreated() {
            context().getEvents().post(new NewSessionCreated());
        }

        @Override
        public void onUpdateReceived(Object obj) {
            if (context().getUpdatesModule() != null) {
                context().getUpdatesModule().onUpdateReceived(obj);
            }
        }

        @Override
        public void onConnectionsChanged(int count) {
            context().getEvents().post(new ConnectingStateChanged(count == 0));
        }
    }
}
