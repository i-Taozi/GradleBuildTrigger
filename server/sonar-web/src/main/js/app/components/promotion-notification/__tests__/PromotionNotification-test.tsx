/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { shallow } from 'enzyme';
import React from 'react';
import { dismissSonarlintAd } from '../../../../api/users';
import { mockCurrentUser, mockLoggedInUser } from '../../../../helpers/testMocks';
import { PromotionNotification, PromotionNotificationProps } from '../PromotionNotification';

jest.mock('../../../../api/users', () => ({
  dismissSonarlintAd: jest.fn()
}));

beforeEach(() => {
  jest.clearAllMocks();
});

it('should render correctly', () => {
  expect(shallowRender()).toMatchSnapshot('anonymous');
  expect(
    shallowRender({ currentUser: mockLoggedInUser({ sonarLintAdSeen: true }) })
  ).toMatchSnapshot('adAlreadySeen');
  expect(shallowRender({ currentUser: mockLoggedInUser() })).toMatchSnapshot('loggedIn');
});

it('should remove the toaster when click on dismiss', () => {
  const setSonarlintAd = jest.fn();
  const wrapper = shallowRender({
    currentUser: mockLoggedInUser({ sonarLintAdSeen: false }),
    setSonarlintAd
  });
  wrapper.find('.toaster-actions ButtonLink').simulate('click');
  expect(dismissSonarlintAd).toBeCalled();
  expect(setSonarlintAd).toBeCalled();
});

it('should remove the toaster and navigate to sonarlint when click on learn more', () => {
  const setSonarlintAd = jest.fn();
  const wrapper = shallowRender({
    currentUser: mockLoggedInUser({ sonarLintAdSeen: false }),
    setSonarlintAd
  });
  wrapper.find('.toaster-actions .button-primary').simulate('click');
  expect(dismissSonarlintAd).toBeCalled();
  expect(setSonarlintAd).toBeCalled();
});

function shallowRender(props: Partial<PromotionNotificationProps> = {}) {
  return shallow(
    <PromotionNotification currentUser={mockCurrentUser()} setSonarlintAd={jest.fn()} {...props} />
  );
}
