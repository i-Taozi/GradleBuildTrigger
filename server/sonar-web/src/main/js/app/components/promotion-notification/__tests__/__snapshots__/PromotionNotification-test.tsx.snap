// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly: adAlreadySeen 1`] = `""`;

exports[`should render correctly: anonymous 1`] = `""`;

exports[`should render correctly: loggedIn 1`] = `
<div
  className="toaster display-flex-center big-padded-left big-padded-right"
>
  <div
    className="toaster-icon spacer-right"
  >
    <img
      alt="SonarQube + SonarLint"
      height={80}
      src="/images/sq-sl.svg"
    />
  </div>
  <div
    className="toaster-content flex-1 padded-left padded-right big-padded-top big-padded-bottom"
  >
    <span
      className="toaster-title text-bold medium"
    >
      promotion.sonarlint.title
    </span>
    <p
      className="spacer-top"
    >
      promotion.sonarlint.content
    </p>
  </div>
  <div
    className="toaster-actions spacer-left padded-left display-flex-column display-flex-center"
  >
    <a
      className="button button-primary big-spacer-bottom"
      href="https://www.sonarqube.org/sonarlint/?referrer=sonarqube-welcome"
      onClick={[Function]}
      rel="noreferrer"
      target="_blank"
    >
      learn_more
    </a>
    <ButtonLink
      className="toaster-link"
      onClick={[Function]}
    >
      dismiss
    </ButtonLink>
  </div>
</div>
`;
